### PR TITLE
fix: add missing write version test

### LINKED ISSUE #88: WriteVersion Test for Maven Project
### Feature Request

Test for writeNextVersion for MavenProjectType

### PR DESCRIPTION
**Pull request by ADI10HERO**
fixes #88 

This PR adds a missing test case for write version for Maven project type 

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Signed-off-by: Aditya Srivastava <adityasrivastava301199@gmail.com>

### CODE REVIEW

File: src/test/java/io/jenkins/plugins/conventionalcommits/utils/WriteVersionTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    pomWriter.write(pomContent);
+    pomWriter.close();
+
+    assertThat(processHelper, is(notNullValue()));
+    when(processHelper.runProcessBuilder(mavenDir, command)).thenReturn("2.0.0");
+
+    WriteVersion writer = new WriteVersion();
+    writer.setProcessHelper(processHelper);
+    writer.write(Version.valueOf("2.0.0"), mavenDir);
+  }
```
Comment (philippart-s) [inline src/test/java/io/jenkins/plugins/conventionalcommits/utils/WriteVersionTest.java:62]: if I can add a suggestion : as you mock your ProcessHelper and use a Maven project to test, perhaps to do a better test you can add a check to the call to the ProcessHelper with right parameters : https://github.com/jenkinsci/conventional-commits-plugin/blob/75c4678aa487f61d1fba7496d5e6086585a1d1dd/src/main/java/io/jenkins/plugins/conventionalcommits/utils/MavenProjectType.java#L50

**Review by philippart-s — COMMENTED** [review]
Just a suggestion to improve (in my opinion :wink: ) yout test

**Review by kwhetstone — COMMENTED** [review]
Can you add more tests?  This looks as if it only covers the case where the write works.  Which is great; we need to confirm that the test is ok!  But what you'll also want to do is test for when things go wrong.  What should the behavior be, and how will it percolate up to the Jenkins build result?

**Comment by ADI10HERO** [comment]
@kwhetstone Gotcha! I will do that 😸

**Review by kwhetstone — APPROVED** [review]
Thanks for adding the test!

**Review by garethjevans — APPROVED** [review]

**Pull request merged**